client/src/components/Auth/RegisterForm.js

```diff
@@ -0,0 +1,395 @@
+import CloseIcon from '@material-ui/icons/Close';
+
+import { dispatchRegister } from '../../store/auth/actions';
+import { dispatchModalClose, dispatchModalOpen } from '../../store/modal/actions';
+import { useStyles } from './styles';
+
+import Tungsten from '../../fonts/Tungsten-Book.woff';
+import MuseoSans from '../../fonts/MuseoSans-500.woff';
+import FacebookSvg from '../../img/auth/facebook.svg';
+import GoogleSvg from '../../img/auth/google.svg';
+
+const tungsten = {
+  fontFamily: 'Tungsten Book',
+  fontStyle: 'normal',
+  src: `
+    local('Tungsten Book'),
+    url(${Tungsten}) format('woff')
+  `,
+};
+
+const museo = {
+  fontFamily: 'Museo Sans 500',
+  fontStyle: 'normal',
+  src: `
+    local('Museo Sans 500'),
+    url(${MuseoSans}) format('woff')
+  `,
+};
+
+
+const useMobileStyles = makeStyles((theme) => ({
+  container: {
+    [theme.breakpoints.down(768)]: {
+      width: '90%',
+    },
+  },
+  title: {
+    [theme.breakpoints.down(768)]: {
+      width: '100%',
+      display: 'flex',
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+    },
+  },
+  separ: {
+    [theme.breakpoints.down(768)]: {
+      marginTop: '25px',
+    },
+  },
+  inactive: {
+    [theme.breakpoints.down(768)]: {
+      color: '#AAAAAA !important',
+      cursor: 'pointer',
+    },
+  },
+  socialIcon: {
+    [theme.breakpoints.down(768)]: {
+      backgroundColor: 'transparent',
+      width: '100%',
+    },
+  },
+}));
+
+
+function RegisterForm(props) {
+  const classes = useStyles();
+  const classesMobile = useMobileStyles();
+
+  const [passwordHidden, setPasswordVisible] = useState(true);
+  const PasswordIcon = passwordHidden ? VisibilityIcon : VisibilityOffIcon;
+  const [state, setState] = useState({
+    email: '',
+    login: '',
+    password: '',
+    repeatPassword: '',
+  });
+
+  const matchMobile = useMediaQuery(props.theme.breakpoints.down(768));
+
+  const validate = () => {
+    const trimmed = Object.keys(state).reduce((trm, key) => {
+      trm[key] = state[key].trim();
+      return trm;
+    }, {});
+
+    if (!trimmed.email || !validator.isEmail(trimmed.email)) {
+      // TODO:  Input notification about error
+      return false;
+    }
+
+    if (!trimmed.login || trimmed.login.length < 4) {
+      // TODO:  Input notification about error
+      return false;
+    }
+
+    if (!trimmed.password || trimmed.password.length < 8) {
+      // TODO:  Input notification about error
+      return false;
+    }
+
+    if (trimmed.password !== trimmed.repeatPassword) {
+      // TODO:  Input notification about error
+      return false;
+    }
+
+    return true;
+  };
+
+  const onChange = (stateName) => (event) => {
+    setState({ ...state, [stateName]: event.target.value });
+  };
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+
+    const { email, login, password } = state;
+
+    if (validate()) {
+      console.log(state, 'REQUEST');
```

Comment:
it's better to comment console.log or delete.

Reply:
resolved